git-ftp

```diff
@@ -811,6 +819,40 @@ handle_remote_protocol_options() {
 	[ "$CURL_INSECURE" != "0" ] && REMOTE_CMD_OPTIONS="$REMOTE_CMD_OPTIONS -k"
 }
 
+extract_folder_snapshot() {
+    local fulladdress=$1
+    SNAPSHOT_DIR="$(basename "$fulladdress")"
+}
+
+save_credential() {
+    git config git-ftp.user $REMOTE_USER
+    git config git-ftp.url $URL
+    git config git-ftp.password $REMOTE_PASSWD
+}
+
+init_new_repository() {
+    
+    [ -z "$URL" ] && print_error_and_die "Error: give a url to snapshot." "$ERROR_USAGE"
+    [ -z $SNAPSHOT_DIR ] && extract_folder_snapshot $URL   # empty string, "clone" into folder with the same name as in the server
+    
+    if [ -d "$SNAPSHOT_DIR" ]; then    # if directory already exists (including . ) check for emptiness
+        if [ "$(ls -A $SNAPSHOT_DIR)" ]; then
+            print_error_and_die "Directorty $SNAPSHOT_DIR is not empty. In order to initialize a git repository the folder has to be empty." "$ERROR_NOT_EMPTY_DIR"
+        fi
+    else
+        mkdir $SNAPSHOT_DIR 2>/dev/null || print_error_and_die "Error creating directory $SNAPSHOT_DIR. Aborting." "$ERROR_MKDIR"
+    fi
+
+    cd $SNAPSHOT_DIR 2>/dev/null || print_error_and_die "Error entering new folder. Aborting." "$ERROR_CDDIR"
+    git init
+}
+
+commit_cloning() {
+    git add . > /dev/null || print_error_and_die "Git: error adding changed files" "$ERROR_GIT"
+    git commit -m "Download $URL with git-ftp" -q  > /dev/null || print_error_and_die "Git: error commiting the changes" "$ERROR_GIT"
+    git ftp -u $REMOTE_USER -p $REMOTE_PASSWD $URL catchup  > /dev/null || print_error_and_die "Uploads current SHA1 to log" "$ERROR_LOG"
```

Comment:
I don't think calling `git ftp` again is a good idea.

- You don't know if there are multiple versions installed on the system and if you are calling the right one.
- You don't include all the possible options like `--ask-passwd`, `--active`, `--dry-run` or `--verbose`.
- It would be more efficient to call the catchup actions within the same script.